src/item_search.cpp

```diff
@@ -43,6 +43,31 @@ std::function<bool( const item & )> basic_item_filter( std::string filter )
                 return item_filter_from_string( pair.first )( i )
                        && item_filter_from_string( pair.second )( i );
             };
+        case 'd'://disassembled components
+            return [filter]( const item & i ) {
+                if( i.components.empty() ) {
+                    //If item wasn't crafted use default recipe components
+                    const auto &req = recipe_dictionary::get_uncraft( i.typeId() ).disassembly_requirements();
+                    if( req.is_empty() ) {
+                        return false;
```

Comment:
Is this early exit really necessary?  Is `req.get_components()` incorrect or expensive to call when `req.is_empty()`?

Reply:
Removed.